lib/cipherstash/index/bloom_filter.rb

```diff
@@ -93,12 +93,20 @@ def to_a
       def add_single_term(term)
         hash = OpenSSL::HMAC.digest("SHA256", @key, term)
 
-        (0..@filter_term_bits-1).map do |slice_idx|
-          slice = hash[2*slice_idx..2*slice_idx+1]
-          bit_position = slice.unpack("S<").first % @filter_size
+        (0..@filter_term_bits-1).map do |slice_index|
+          byte_slice = two_byte_slice(hash, slice_index)
+          bit_position = little_endian_uint16_from_byte_slice(byte_slice) % @filter_size
```

Comment:
Nice. Much clearer! Probably slower due to function calls on the stack but compared to how slow Ruby will be it doesn't matter. Rust would likely optimise it away anyway.